apps/core/scenes/bookingDetail/sectorDetails/segments/SegmentMap.js

```diff
@@ -6,50 +6,117 @@ import { StyleSheet } from '@kiwicom/universal-components';
 import { graphql, createFragmentContainer } from '@kiwicom/margarita-relay';
 import { getCenter } from 'geolib';
 import memoize from 'memoize-one';
-import stringify from 'json-stable-stringify';
+import { uniqBy, prop } from 'ramda';
 
 import type { SegmentMap as BookingType } from './__generated__/SegmentMap.graphql';
 import MapLines from './MapLines';
+import SegmentMapMarker from './SegmentMapMarker';
 
 type Props = {|
   +data: ?BookingType,
 |};
 
+type RouteStop = {|
+  +stop: ?{|
+    +city: ?{|
+      +name: ?string,
+    |},
+    +locationId: ?string,
+    +coordinates: ?{|
+      +latitude: ?number,
+      +longitude: ?number,
+    |},
+  |},
+|};
+
+type Segments = ?$ReadOnlyArray<?{|
+  +arrival: ?RouteStop,
+  +departure: ?RouteStop,
+|}>;
+
+type MarkerData = {|
+  +latitude: ?number,
+  +longitude: ?number,
+  +name: ?string,
+  +locationId: string,
+|};
+
 const DEFAULT_PADDING = { top: 40, right: 40, bottom: 40, left: 40 };
 
-const mapCoordinates = memoize(
-  (segmentLocations: ?$PropertyType<BookingType, 'segmentLocations'>) => {
-    const coordinates = segmentLocations ?? [];
-    return coordinates.map(coordinate => ({
-      latitude: coordinate?.lat,
-      longitude: coordinate?.lng,
-    }));
-  },
-);
+const mapMarkerData = memoize((props: Props) => {
+  switch (props.data?.type) {
+    case 'BOOKING_ONE_WAY':
+      return mapSegmentsToMarkerData(props.data?.sector?.segments);
```

Comment:
Just to be uniform, could you extract this as a function `getOneWaySegments`?

Reply:
I don't think that makes sense, because there are no computations to be done. I don't need a function to return `props.data?.sector?.segments`